### PR TITLE
Feature/save data item api

### PR DESCRIPTION
**Pull request by elisehuard**
- when upload is finished on the front-end, this is the api to call to save the details of the data item.
- edits to get the upload to work with the newer signature version (V4) for AWS s3

### CODE REVIEW

File: src/witan/app/forecast.clj [hunk 1]
```diff
@@ -340,3 +361,31 @@
...
+  :exists? (fn [ctx]
+             (let [forecast (get-forecast-version id version)
+                   category-exists (some->> forecast
+                                            :model_id
+                                            (model/get-model-by-model-id)
+                                            :input_data
+                                            (some #{category}))]
+               [category-exists {:forecast forecast}]))
+  :processable? (fn [ctx]
+                  (and (util/post!-processable-validation ws/NewDataItem)
```
Comment (acron0) [inline src/witan/app/forecast.clj]: `util/post!-processable-validation` actually returns a fn so this line should be `((util/post!-processable-validation ws/NewDataItem) ctx)` - I do not recall why it was written this way....

File: src/witan/app/data.clj [hunk 2]
```diff
@@ -26,26 +29,43 @@
...
-                               :name name
-                               :publisher publisher
-                               :version version
-                               :s3-url s3-url} %)) '(:data_by_data_id :data_by_category))
-  (first (c/exec (find-data-by-data-id data-id))))
+  "add data version"
+  [{:keys [data-id category name file-name publisher s3-key]
+    :or {data-id (uuid/random)}}]
+  (let [version (or (get-current-version-name name) 1)]
+    (mapv #(c/exec (create-data {:data-id data-id
```
Comment (acron0) [inline src/witan/app/data.clj]: You're not using the result of `mapv` so would `doseq` be better?

Comment (sw1nn) [inline src/witan/app/data.clj]: clojure 1.7.0 has `run!`, which can be better than `doseq` for this specific use case,

**Comment by acron0** [comment]
:shipit:

**Pull request merged**